### PR TITLE
[Upgrade] Add Initial Part of Upgrade to Sylius 2.0

### PR DESCRIPTION
**Pull request by Rafikooo**
| Q               | A
|-----------------|-----
| Branch?         | 2.0 
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no 
| License         | MIT

### CODE REVIEW

File: UPGRADE-2.0.md [hunk 1]
```diff
@@ -1,17 +1,92 @@
...
+
+```md
+#config/routes/sylius_shop.yaml
+
+sylius_shop_payum:
+-   resource: "@SyliusShopBundle/Resources/config/routing/payum.yml"
++   resource: "@SyliusPayumBundle/Resources/config/routing/integrations/sylius_shop.yaml"
+
+sylius_payment_notify:
++   resource: "@SyliusPaymentBundle/Resources/config/routing/integrations/sylius.yaml"
```
Comment (Rafikooo) [inline UPGRADE-2.0.md]: I’m not sure whether it should be configured in the `config/routes/sylius_shop.yaml` shop file or in a separate one.

Comment (NoResponseMate) [reply UPGRADE-2.0.md]: Can be in the `sylius_shop` routing file, **but** should not be localised (since that'd make the route unnecessarily stateful), it might be a good idea to add this info here (same deal w/ payum routes).

File: UPGRADE-2.0.md [hunk 2]
```diff
@@ -1,5 +1,126 @@
...
+    resource: "@SyliusApiBundle/Resources/config/routing.yml"
+-   prefix: "%sylius.security.new_api_route%"
++   prefix: "%sylius.security.api_route%"
+
+```
+
+* Routing changes (note that these shop routes are not localized with the prefix: /{_locale} configuration entry):
+
+```md
+#config/routes/sylius_shop.yaml
```
Comment (NoResponseMate) [inline UPGRADE-2.0.md:74]:
```suggestion
``` diff
# config/routes/sylius_shop.yaml
```

File: UPGRADE-2.0.md [hunk 3]
```diff
@@ -1,5 +1,126 @@
...
+
+sylius_payment_notify:
++   resource: "@SyliusPaymentBundle/Resources/config/routing/integrations/sylius.yaml"
+
+```
+
+* Bundle configuration changes:
+
+```md
+#config/bundles.php
```
Comment (NoResponseMate) [inline UPGRADE-2.0.md:88]:
```suggestion
```diff
# config/bundles.php
```

File: UPGRADE-2.0.md [hunk 4]
```diff
@@ -1,5 +1,126 @@
...
+
+* New Symfony/Messenger transports for handling payment requests have been added. 
+Therefore, you need to add the following configuration to your .env file:
+```md
+###> symfony/messenger ###
+...
+SYLIUS_MESSENGER_TRANSPORT_PAYMENT_REQUEST_DSN=doctrine://default?queue_name=payment_request
+SYLIUS_MESSENGER_TRANSPORT_PAYMENT_REQUEST_FAILED_DSN=doctrine://default?queue_name=payment_request_failed
+###< symfony/messenger ###
+```
```
Comment (NoResponseMate) [inline UPGRADE-2.0.md:116]:
```suggestion
Therefore, you need to add the following configuration to your .env file:

```diff
###> symfony/messenger ###
...
+ SYLIUS_MESSENGER_TRANSPORT_PAYMENT_REQUEST_DSN=doctrine://default?queue_name=payment_request
+ SYLIUS_MESSENGER_TRANSPORT_PAYMENT_REQUEST_FAILED_DSN=doctrine://default?queue_name=payment_request_failed
###< symfony/messenger ###
```
```

File: UPGRADE-2.0.md [hunk 5]
```diff
@@ -1208,16 +1339,21 @@ In modern Symfony projects, the hasher name is stored on the password.
 Example:
 `$argon2i$v=19$m=65536,t=4,p=1$VVJuMnpUUWhRY1daN1ppMA$2Tx6l3I+OUx+PUPn+vZz1jI3Z6l6IHh2kpG0NdpmYWE`
 
-If some of your users do not have the hasher name stored in the password field you may need to configure the "migrate_from" option into Symfony, following that documentation:
+If some of your users do not have the hasher name stored in the password field you may need to configure the "
+migrate_from" option into Symfony, following that documentation:
```
Comment (NoResponseMate) [inline UPGRADE-2.0.md:1343]:
```suggestion
If some of your users do not have the hasher name stored in the password field you may need to configure the
"migrate_from" option by following the documentation:
```

File: UPGRADE-2.0.md [hunk 6]
```diff
@@ -1433,13 +1569,16 @@ If your app never changed the hasher name configuration, you don't need to confi
 ## Payment method gateways
 
 * Stripe gateway has been removed. This implementation has been deprecated and not SCA Ready.
-* PayPal Express Checkout gateway has been removed. Use now [PayPal Commerce Platform](https://github.com/Sylius/PayPalPlugin) integration.
+* PayPal Express Checkout gateway has been removed. Use
+  now [PayPal Commerce Platform](https://github.com/Sylius/PayPalPlugin) integration.
```
Comment (NoResponseMate) [inline UPGRADE-2.0.md:1573]:
```suggestion
* PayPal Express Checkout gateway has been removed.
  Use the [PayPal Commerce Platform](https://github.com/Sylius/PayPalPlugin) integration.
```

File: UPGRADE-2.0.md [hunk 7]
```diff
@@ -1433,13 +1569,16 @@ If your app never changed the hasher name configuration, you don't need to confi
...
 
 ## Theming
-* Dependency on `sylius/theme-bundle` is moved from CoreBundle to ShopBundle and it will no longer be installed 
+
+* Dependency on `sylius/theme-bundle` is moved from CoreBundle to ShopBundle and it will no longer be installed
```
Comment (NoResponseMate) [inline UPGRADE-2.0.md:1577]:
```suggestion
* The `sylius/theme-bundle` dependency has been moved from CoreBundle to ShopBundle, so it will no longer be installed
```

File: UPGRADE-2.0.md [hunk 8]
```diff
@@ -1,5 +1,126 @@
...
+## Initial part
+
+Even if your app is barely customized, it will require some manual adjustments before it can run again. Depending on
+whether you use Symfony Flex, some of these changes may be applied automatically, but it’s important to check them
+manually regardless.
+
+* Packages configuration changes:
+
+```md
+#config/packages/_sylius.yaml
```
Comment (NoResponseMate) [inline UPGRADE-2.0.md:12]:
```suggestion
```diff
# config/packages/_sylius.yaml
```

File: UPGRADE-2.0.md [hunk 9]
```diff
@@ -1,5 +1,126 @@
...
+
+sylius_payum:
+    resources:
+-       gateway_config:
+-           classes:
+-               model: App\Entity\Payment\GatewayConfig
+```
+
+* API firewalls have been renamed and user checkers have been configured on firewalls
+  with `security.user_checker.chain.<firewall>` service:
```
Comment (NoResponseMate) [inline UPGRADE-2.0.md:36]:
```suggestion
* API firewalls have been renamed and user checkers have been configured on
  them with `security.user_checker.chain.<firewall>` services:
```

File: UPGRADE-2.0.md [hunk 10]
```diff
@@ -1,5 +1,126 @@
...
+
+```diff
+# config/routes/sylius_api.yaml
+
+sylius_api:
+    resource: "@SyliusApiBundle/Resources/config/routing.yml"
+-   prefix: "%sylius.security.new_api_route%"
++   prefix: "%sylius.security.api_route%"
+
+```
```
Comment (NoResponseMate) [inline UPGRADE-2.0.md:69]: This is not related to the security configuration above, it's a change of parameters and fits more under `routing changes`

File: UPGRADE-2.0.md [hunk 11]
```diff
@@ -1,5 +1,126 @@
...
++   Symfony\UX\Autocomplete\AutocompleteBundle::class => ['all' => true],
+];
+
+* New Symfony/Messenger transports for handling payment requests have been added. 
+Therefore, you need to add the following configuration to your .env file:
+```md
+###> symfony/messenger ###
+...
+SYLIUS_MESSENGER_TRANSPORT_PAYMENT_REQUEST_DSN=doctrine://default?queue_name=payment_request
+SYLIUS_MESSENGER_TRANSPORT_PAYMENT_REQUEST_FAILED_DSN=doctrine://default?queue_name=payment_request_failed
```
Comment (Prometee) [inline UPGRADE-2.0.md:114]:
```suggestion
SYLIUS_MESSENGER_TRANSPORT_PAYMENT_REQUEST_DSN=sync://
SYLIUS_MESSENGER_TRANSPORT_PAYMENT_REQUEST_FAILED_DSN=sync://
```
Async payment request can't be supported by Sylius right now, I made a PR (already merged) about this default value.

**Comment by TheMilek** [comment]
Closing, as it is continued in https://github.com/Sylius/Sylius/pull/17448 with resolved review :v:
